solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+    */
+    function connectorTokens(uint256 _index) public view override returns (IERC20Token) {
+        return reserveTokens[_index];
+    }
+
+    /**
+      * @dev deprecated, backward compatibility
+    */
+    function connectorTokenCount() public view override returns (uint16) {
+        return reserveTokenCount();
+    }
+
+    /**
+      * @dev deprecated, backward compatibility
+    */
+    function getConnectorBalance(IERC20Token _connectorToken) public view override returns (uint256) {
+        return reserveBalance(_connectorToken);
+    }
+
+    /**
+      * @dev deprecated, backward compatibility
+    */
+    function getReturn(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount) public view returns (uint256, uint256) {
+        return targetAmountAndFee(_sourceToken, _targetToken, _amount);
+    }
+
+    /**
+      * @dev returns the converter type
+      *
+      * @return see the converter types in the the main contract doc
+    */
+    function converterType() public pure override returns (uint16) {
+        return 3;
+    }
+
+    /**
+      * @dev returns the expected target amount of converting one reserve to another along with the fee
+      *
+      * @param _sourceToken contract address of the source reserve token
+      * @param _targetToken contract address of the target reserve token
+      * @param _amount      amount of tokens received from the user
+      *
+      * @return expected target amount
+      * @return expected fee
+    */
+    function targetAmountAndFee(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount)
+        public
+        view
+        override
+        active
+        returns (uint256, uint256)
+    {
+        uint256 sourceId = reserveIds[_sourceToken];
+        uint256 targetId = reserveIds[_targetToken];
+
+        (uint256 sourceBalance, uint256 targetBalance) = getReserveBalances(sourceId, targetId);
+        uint256 amount = crossReserveTargetAmount(sourceBalance, targetBalance, _amount);
+
+        // return the amount minus the conversion fee and the conversion fee
+        uint256 fee = calculateFee(amount);
+        return (amount - fee, fee);
+    }
+
+    /**
+      * @dev converts a specific amount of source tokens to target tokens
+      *
+      * @param _sourceToken source ERC20 token
+      * @param _targetToken target ERC20 token
+      * @param _amount      amount of tokens to convert (in units of the source token)
+      * @param _trader      address of the caller who executed the conversion
+      * @param _beneficiary wallet to receive the conversion result
+      *
+      * @return amount of tokens received (in units of the target token)
+    */
+    function doConvert(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount, address _trader, address payable _beneficiary)
+        internal
+        returns (uint256)
+    {
+        // update the recent average rate
+        updateRecentAverageRateIfNeeded();
```

Comment:
no need for `IfNeeded` in the name (not to mention it's not this function's responsibility to know) - the `updateRecentAverageRate` function owns it all, it just needs to be called.

Reply:
Done